### ISSUE TITLE
Glue returned `error` for statement None for code SqlWrapper2.execute()

### ISSUE THREAD

**Issue by dboyadzhiev** [aws-samples/dbt-glue#77]
### Describe the bug
I tried to set up a simple project to explore dbt+glue: read and write from/to parquet file. Run `dbt run`. But there is an error.

### Steps To Reproduce
- create an S3 bucket with parquet files
- defined a source database/table and run a crawler in DataCatalog 
- create an IAM role with permissions (described in the documentation)
- create a target s3 bucket where to save the output files
- create a destination database in DataCatalog 

### Expected behavior
To create the expected s3 files

### Screenshots and log output
```
14:04:36  Running with dbt=1.2.1
14:04:36  Unable to do partial parsing because profile has changed
14:04:36  Found 1 model, 0 tests, 0 snapshots, 0 analyses, 323 macros, 0 operations, 0 seed files, 1 source, 0 exposures, 0 metrics
14:04:36  
14:05:19  Concurrency: 1 threads (target='dev')
14:05:19  
14:05:19  1 of 1 START table model dbt_que_demo.my_first_dbt_model ....................... [RUN]

Glue adapter: Glue returned `error` for statement None for code SqlWrapper2.execute('''
    create table dbt_que_demo.my_first_dbt_model
    using parquet
    LOCATION 's3://demo-bucket-dai/dbt_que_demo/my_first_dbt_model/'
    as
    select tenant, id, datetime_gmt, report_type
    from glue_source_db.reports
    limit 10
'''), 

Py4JError: An error occurred while calling o91.toString. Trace:
java.lang.IllegalArgumentException: object is not an instance of declaring class
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.lang.Thread.run(Thread.java:750)

14:05:27  1 of 1 ERROR creating table model dbt_que_demo.my_first_dbt_model .............. [ERROR in 7.57s]
```

### System information
**The output of `dbt --version`:**
```
Core:
  - installed: 1.2.1
  - latest:    1.2.1 - Up to date!

Plugins:
  - spark: 1.2.0 - Up to date!
```

**The operating system you're using:** 
```
System Version: macOS 12.4 (21F79)
Kernel Version: Darwin 21.5.0
Chip: Apple M1
```

**The output of `python --version`:** 
```3.9.14```

**The output of `pip list`:** 
[github gist](https://gist.github.com/dboyadzhiev/bd1f58c28f32ed563f8c68fe80250682.js)

### Additional context
None

**Comment by sinogermany** [comment]
After quite a bit of digging - here's what worked for me:

TLDR:
- Downgrade the pip package versions
- Update the IAM policy

First thing first - the versions worked for me were:
```
dbt-core==1.1.2
dbt-glue==0.2.1
```

The IAM Policy has a few changes:
```json
"Effect": "Allow",
"Principal": {
  "Service": [
    "glue.amazonaws.com",
    "elasticmapreduce.amazonaws.com"
  ]
},
```

Also, for the `GetObject` action, we need to add `/*` to the end:
```json
"Resource": [
  "arn:aws:s3:::bucket-name/*"
],
```
instead of
```json
"Resource": [
  "arn:aws:s3:::bucket-name"
],
```

After all those changes, the models ran fine. Hope that save some debugging time for some friends.

**Comment by nicor88** [comment]
This issue is most likely due to the fact that in glue 3.0 another iam policy is needed, specifically `glue:GetUserDefinedFunctions`. I found out running the examples dbt-repo in a glue-session via a notebook, first using glue 2.0 (all worked fine), then using glue 3.0, where I got an access denied error.

Note: This IAM action is not needed if you use glue 2.0. @dboyadzhiev please confirm the above. I tested multiple times, and now all works for me.

**Comment by dboyadzhiev** [comment]
Yes @nicor88, when I explicitly set the glue-adapter's version to `2.0` the error disappeared.

**Comment by nicor88** [comment]
@dboyadzhiev as mentioned here https://github.com/aws-samples/dbtgluenyctaxidemo/pull/1 consider extending your IAM policy and add `glue:GetUserDefinedFunctions`, doing so I was able to make the adapter work with glue 3.0.
